spicy/runtime/src/parser.cc

```diff
@@ -17,6 +18,16 @@ HILTI_EXCEPTION_IMPL(Backtrack)
 HILTI_EXCEPTION_IMPL(MissingData);
 HILTI_EXCEPTION_IMPL(ParseError)
 
+void spicy::rt::confirm_input() {
+    if ( const auto& hook = globalState()->configuration->hook_confirm_input )
```

Comment:
I'm wondering if it would be possible/easy to pass more context like the active `ParsedUnit` and maybe even `Parser` instance?

I once tried to write a host application and am wondering how I could, for example, print info about the unit or parser without needing to reach to static / globals? And seems spicy might have that info and could inject it?

Reply:
(And no, I don't have a direct use case, was just curious)

Reply:
Similar answer unfortunately than above to Benjamin's suggestion: we don't have that information available because it's the user calling our new functions; we only have access to whatever is passed as parameters there. 

However, in this case I think there is an even better salutation: the host application can track whatever state it needs, and it will be accessible by the callback implementations through the current `cookie`. That's for example how the Zeek plugin knows what the current connection/file/packet is. This is "even better" I think because we don't need to hardcode some specific choice of what to make available on the Spicy side.

Reply:
I guess I'll need to look at the spicy-plugin :-)

Probably `std::function` can just hold a callable object, too, so that getting back to the cookie or some other state doesn't necessarily mean using global variables.